packages/link/components/UnstyledLink.tsx

```diff
@@ -1,32 +1,23 @@
 import React from "react";
 import { ExpandedLinkProps } from "../types";
-import { LinkComponentContext } from "../../uiKitProvider/link/context";
 
-class UnstyledLink extends React.Component<ExpandedLinkProps> {
-  static contextType = LinkComponentContext;
-  render() {
-    const LinkComponent = this.context;
+const UnstyledLink = (props: ExpandedLinkProps) => {
+  const { href, url, target, openInNewTab, context, children, ...rest } = props;
 
-    if (LinkComponent) {
-      return <LinkComponent {...this.props} />;
-    }
+  // Context is used for `UIKitProvider` link delegation
+  const LinkComponent = context;
```

Comment:
try `const LinkComponent = React.useContext(LinkComponentContext)` here and remove context from props.

Reply:
Thanks for taking a look at this. Looks like this does the trick here 👍🏻